containers/ReferenceValue.js

```diff
@@ -0,0 +1,16 @@
+import React from 'react'
+import { connect } from 'react-redux'
+import { changeReferenceValue } from '../actions'
+
+let ReferenceValue = ({ entityId, refId, referenceValue, dispatch }) => (
+  <div>
+    <input type="text" 
+      placeholder="reference value"
+      value={referenceValue}
+      onChange={e => dispatch(changeReferenceValue(entityId, refId, e.target.value))}/>
```

Comment:
I'll see if I can put them in components, but I don't have a clear solution for the above potential problem I mentioned.